cdc/entry/schema_storage.go

```diff
@@ -192,13 +192,8 @@ func (s *schemaStorageImpl) HandleDDLJob(job *timodel.Job) error {
 
 // AdvanceResolvedTs advances the resolved
 func (s *schemaStorageImpl) AdvanceResolvedTs(ts uint64) {
-	var swapped bool
-	for !swapped {
-		oldResolvedTs := atomic.LoadUint64(&s.resolvedTs)
-		if ts < oldResolvedTs {
-			return
-		}
-		swapped = atomic.CompareAndSwapUint64(&s.resolvedTs, oldResolvedTs, ts)
+	if ts > s.ResolvedTs() {
+		atomic.StoreUint64(&s.resolvedTs, ts)
 	}
```

Comment:
The optimization is correct only if `AdvanceResolvedTs` won't be called concurrently. Please add a comment about it.

Otherwise there will be write skew such as
- ts1 > ts2 > s.ResolvedTs
- Both `AdvanceResolvedTs(ts1)` and `AdvanceResolvedTs(ts2)` enters L196
- The value of `s.resolvedTs` will be either `ts1` or `ts`, which is not expected.

Reply:
Done. It is optimized since we don't need to call it concurrently.